PC/pyconfig.h.in

```diff
@@ -311,9 +311,13 @@ Py_NO_ENABLE_SHARED to find out.  Also support MS_NO_COREDLL for b/w compat */
 #ifdef MS_COREDLL
 #       if !defined(Py_BUILD_CORE) && !defined(Py_BUILD_CORE_BUILTIN)
                 /* not building the core - must be an ext */
-#               if defined(_MSC_VER)
+#               if defined(_MSC_VER) && !defined(Py_NO_LINK_LIB)
```

Comment:
This whole set of conditions is pretty ugly... any interest in simplifying?

Ideally in this order:
* check if the feature is explicitly disabled (new variable)
* check if we're in a context where it should be implicitly disabled (`Py_BUILD_CORE[_BUILTIN]`)
* check if the compiler supports it (`_MSC_VER`)

The order doesn't need to be spelled out to anyone reading it, but it should make it easiest to read (the first check has the best name) and easiest to maintain (alternate compilers can be supported in the "middle", not the edges).

It also really doesn't need to be indented by 8 characters. Four would be plenty.